connector/authproxy/authproxy.go

```diff
@@ -15,7 +15,8 @@ import (
 // Config holds the configuration parameters for a connector which returns an
 // identity with the HTTP header X-Remote-User as verified email.
 type Config struct {
-	UserHeader string `json:"userHeader"`
+	UserHeader string   `json:"userHeader"`
+	Groups     []string `json:"groups"`
```

Comment:
From my point of view, it would be better (and more generic) to extract groups from headers (as we do for username).

For example, the oauth2 proxy can return groups via the `X-Forwarded-Groups` header.

We even have a TODO in code to implement groups support this way:
https://github.com/dexidp/dex/blob/716eef83bca805e1607338fe0a271affc2d097b0/connector/authproxy/authproxy.go#L58

WDYT? Will extracting groups from headers cover your use case?

Reply:
Maybe we can do both ? My auth proxy (traefik-forward-auth) does not return `X-Forwarded-Group` Header.
So if the header is present we use it else we fallback to the configured group.

Reply:
I've updated the PR to allow custom group Header configuration and use both header and configured groups (appended)

Reply:
Hello @nabokihms What do you think of my changes ?
default groups are the configured groups (can be empty)
if http header (default `X-Remote-Group` but configurable) is present append it to the configured groups.
I've tested it on my argocd dex deployment, and it works fine.